File: python/collatz-conjecture/collatz_conjecture.py
def steps(n: int) -> int:
    c = 0
    if n <= 0:
        raise ValueError("Only positive integers are allowed")

    while n > 1:
        if n % 2 == 0:
            n = n // 2
            c += 1
        else:
            n = (3 * n) + 1
            c += 1

    return c

File: nim/collatz-conjecture/collatz_conjecture.nim
proc steps*(n: int, c: int = 0): int | ValueError =
   if n <= 0: raise newException(ValueError, "Only positive integers allowed!")
   if n == 1: return c
   elif n mod 2 == 0:
      steps(int(n / 2), c + 1)
   else:
      steps(((n * 3) + 1), c + 1)